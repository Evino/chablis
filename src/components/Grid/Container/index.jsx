import React, { PureComponent } from 'react';
import PropTypes from 'prop-types';
import styled, { css } from 'styled-components';
import { breakpoints } from '../../../styles/variables';

const media = Object.keys(breakpoints).reduce((prev, next) => {
  const breakpoint = `
    @media (min-width: ${breakpoints[next]}) {
      max-width: ${breakpoints[next]};
    }
  `;

  let newValue = prev;
  newValue += breakpoint;

  return newValue;
}, '');


const ContainerDiv = styled.div`
  width: 100%;
  margin: 0 auto;
  box-sizing: border-box;
  transition: .3s ease-out;

  ${media}

  ${props => props.full && css`
    max-width: 100% !important;
  `}
`;

export default class Container extends PureComponent {
  static propTypes = {
    children: PropTypes.node.isRequired,
    full: PropTypes.bool,
  }

  static defaultProps = {
    full: false,
  }

  render() {
    return (
      <ContainerDiv {...this.props}>{this.props.children}</ContainerDiv>
    );
  }
}
