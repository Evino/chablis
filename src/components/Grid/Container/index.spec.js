import React from 'react';
import { mount } from 'enzyme';
import renderer from 'react-test-renderer';
import Container from '.';

const element = <Container><span>Subcomponent</span></Container>;
const component = mount(element);

describe('Container', () => {
  it('should render a container div', () => {
    const tree = renderer
      .create(element)
      .toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('should render children properly', () => {
    const child = component.find('span');
    expect(child).toHaveLength(1);
  });

  it('should render the full version of container', () => {
    component.setProps({ full: true });
    expect(component.prop('full')).toBeTruthy();
  });

  it('should render the full version of container in some breakpoints', () => {
    component.setProps({ full: false, fullTablet: true });
    expect(component.prop('fullTablet')).toBeTruthy();
  });
});
