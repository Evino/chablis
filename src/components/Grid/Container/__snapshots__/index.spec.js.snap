// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Container should render a container div 1`] = `
<div
  className="Container__ContainerDiv-sc-12ylvnx-0 dIJERq"
>
  <span>
    Subcomponent
  </span>
</div>
`;
